### PR TITLE
sshfsFuse: 2.5 -> 2.7

### PR DESCRIPTION
**Pull request by k0ral**
###### Things done:
- [ ] Tested via `nix.useChroot`.
- [x] Built on platform(s): linux
- [ ] Tested compilation of all pkgs that depend on this change.
- [x] Tested execution of binary products.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### Extra

cc @jgeerds

### CODE REVIEW

File: pkgs/tools/filesystems/sshfs-fuse/default.nix [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
-    sha256 = "0gp6qr33l2p0964j0kds0dfmvyyf5lpgsn11daf0n5fhwm9185z9";
+  src = fetchFromGitHub {
+    repo = "sshfs";
+    owner = "libfuse";
+    rev = "sshfs_2_6";
+    sha256 = "08ffvviinjf8ncs8z494q739a8lky9z46i09ghj1y38qzgvk3fpw";
   };
   
-  buildInputs = [ pkgconfig glib fuse ];
+  buildInputs = [ pkgconfig glib fuse autoconf automake ];
```
Comment (globin) [inline pkgs/tools/filesystems/sshfs-fuse/default.nix]: please use the `autoreconfHook`

**Comment by joachifm** [comment]
Ping @k0ral

**Comment by k0ral** [comment]
Updated using 'autoreconfHook', sorry for the delay.

**Comment by joachifm** [comment]
Looks like there's a new release (2.7) at https://github.com/libfuse/sshfs/releases

**Comment by k0ral** [comment]
Upgraded to 2.7 .

**Comment by joachifm** [comment]
The name and src.rev disagree. Also, is there a reason for building from git rather than using a release tarball? Otherwise, this builds & works for me.

**Comment by k0ral** [comment]
Factored out a `version` attribute as suggested.
Unless I'm mistaken, `fetchFromGitHub` actually downloads a release tarball at `https://github.com/${owner}/${repo}/archive/${rev}.tar.gz`.

**Comment by joachifm** [comment]
Ah, you're right. TIL. Thanks!

**Pull request merged**